### PR TITLE
Adapt for jar scan

### PR DESCRIPTION
**Pull request by wp0pw**
changes to allow for scanning of java archives
- now adds missing components to java sboms
- added `-n` option that does not upload and outputs sbom to stdout instead

### CODE REVIEW

File: scripts/sbom_scraper.sh [hunk 1]
```diff
@@ -213,11 +217,17 @@ then
     popd > /dev/null
 
     COMPONENT_NAME=$(xq -r .bom.metadata.component.name "$OUTPUT")
-    COMPONENT_VERSION=$(xq -r .bom.components.component.version  "${OUTPUT}")
-    ORIG_COMPONENT_NAME="${COMPONENT_NAME}"
-    ORIG_COMPONENT_VERSION="${COMPONENT_VERSION}"
+    COMPONENT_VERSION=$(xq -r .bom.metadata.component.version  "${OUTPUT}")
```
Comment (eccles) [inline scripts/sbom_scraper.sh:220]: nice spot

File: scripts/sbom_scraper.sh [hunk 2]
```diff
@@ -402,67 +477,73 @@ END
 # xmllint complains about a double import of the spdx schema, but we have to import via
 # the wrapper to set the schema location to a local file, as xmllint fails to download
 # them from the internet as they are https
-xmllint "$PATCHED_OUTPUT" --schema "$SCRIPTDIR"/cyclonedx-wrapper.xsd --noout 2>&1 | grep -Fv "Skipping import of schema located at 'http://cyclonedx.org/schema/spdx' for the namespace 'http://cyclonedx.org/schema/spdx'"
-[ "${PIPESTATUS[0]}" -ne 0 ] && exit "${PIPESTATUS[0]}"
+_=$(xmllint "$PATCHED_OUTPUT" --schema "$SCRIPTDIR"/cyclonedx-wrapper.xsd --noout 2>&1 | grep -Fv "Skipping import of schema located at 'http://cyclonedx.org/schema/spdx' for the namespace 'http://cyclonedx.org/schema/spdx'")
```
Comment (eccles) [inline scripts/sbom_scraper.sh:480]: I am wondering do we need this now that the sboms endpoint does proper validation?
Removing the xmllint stuff would make the script simpler and easier to 'sell' to a customer?

File: scripts/sbom_scraper.sh [hunk 3]
```diff
@@ -402,67 +477,73 @@ END
...
+        --data-urlencode "client_secret=${SECRET}" \
+        "${URL}/archivist/iam/v1/appidp/token")
+    if [ "${HTTP_STATUS}" != "200" ]
+    then
+        log "Get token failure ${HTTP_STATUS}"
+        exit 2
+    fi
 
-TOKEN=$(jq -r .access_token "${TEMPDIR}"/access_token )
+    TOKEN=$(jq -r .access_token "${TEMPDIR}"/access_token )
```
Comment (eccles) [inline scripts/sbom_scraper.sh:506]: is there some mixed tab/spaces here ?

Comment (wp0pw) [reply scripts/sbom_scraper.sh:506]: ah, no it's all space - I just indented the whole block as it only happens if we run without -n

File: scripts/sbom_scraper.sh (hunk 2 above)
Comment (wp0pw) [reply scripts/sbom_scraper.sh:480]: I think it's still useful especially considering I added `-n` option for when you don't want to upload the SBOM

**Review by eccles — APPROVED** [review]

**Pull request merged**